legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperModelBuilder.java

```diff
@@ -279,10 +271,10 @@ public static String getTerseSignature(Function function)
 
     private static String terseSignatureSuffix(Function function)
     {
-        return "_" + LazyIterate.collect(function.parameters, HelperModelBuilder::getParameterSignature).select(Objects::nonNull).makeString("__")
+        String functionSignature =  LazyIterate.collect(function.parameters, HelperModelBuilder::getParameterSignature).select(Objects::nonNull).makeString("__")
```

Comment:
Could we make the logic more clear here. Maybe use string builders. Also lets add tests for these functions to make it clear what the expectation that each returns.

Reply:
tests are added